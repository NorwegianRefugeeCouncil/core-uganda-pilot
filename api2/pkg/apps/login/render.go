package login

import "net/http"

func (s *Server) Render(w http.ResponseWriter, req *http.Request, templateName string, data map[string]interface{}) {
	if err := s.template.ExecuteTemplate(w, templateName, data); err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}
