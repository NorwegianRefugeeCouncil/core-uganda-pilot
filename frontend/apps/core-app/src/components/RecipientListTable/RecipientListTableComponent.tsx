import React, { useContext } from 'react';
import { Box, HStack, VStack, FlatList, ScrollView } from 'native-base';
import { Row } from 'react-table';

import { RecipientListTableContext } from './RecipientListTableContext';
import { RecipientListTableRow } from './RecipientListTableRow';
import { RecipientListTableHeaderCell } from './RecipientListTableHeaderCell';

type Props = {
  onItemClick: (id: string) => void;
};

export const RecipientListTableComponent: React.FC<Props> = ({
  onItemClick,
}) => {
  const tableContext = useContext(RecipientListTableContext);
  if (!tableContext) return null;

  const { tableInstance } = tableContext;
  if (!tableInstance) return null;

  const { rows, columns, prepareRow } = tableInstance;

  const renderRow = ({ item }: { item: Row }) => {
    prepareRow(item);
    return (
      <RecipientListTableRow
        key={item.id}
        row={item}
        onRowClick={onItemClick}
      />
    );
  };

  return (
    <ScrollView
      horizontal
      contentContainerStyle={{
        flexGrow: 1,
      }}
    >
      <Box width="100%">
        <HStack>
          {columns.map((column) => (
            <RecipientListTableHeaderCell column={column} key={column.id} />
          ))}
        </HStack>
        <VStack>
          <FlatList data={rows} renderItem={renderRow} />
        </VStack>
      </Box>
    </ScrollView>
  );
};
