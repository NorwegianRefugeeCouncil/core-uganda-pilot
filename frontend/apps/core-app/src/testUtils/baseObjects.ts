import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
