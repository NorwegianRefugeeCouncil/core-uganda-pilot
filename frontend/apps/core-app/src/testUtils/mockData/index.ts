export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
