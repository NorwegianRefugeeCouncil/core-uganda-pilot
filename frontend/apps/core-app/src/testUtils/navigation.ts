export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
