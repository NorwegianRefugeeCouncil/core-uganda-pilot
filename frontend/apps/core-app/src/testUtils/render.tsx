import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
