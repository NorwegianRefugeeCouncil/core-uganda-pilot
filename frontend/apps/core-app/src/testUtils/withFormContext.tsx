import { Button } from 'native-base';
import React from 'react';
import { FieldValues, FormProvider, useForm } from 'react-hook-form';

export const withFormContext =
  <T,>(
    Component: React.FC<T>,
    defaultValues: FieldValues = {},
    onSubmit: (data: any) => void = () => {},
  ) =>
  // eslint-disable-next-line react/display-name
  (props: T) => {
    const f = useForm({ defaultValues });

    const handleSubmit = f.handleSubmit((data: any) => {
      onSubmit(data);
    });

    return (
      <FormProvider {...f}>
        <Component {...props} />
        <Button testID="with-form-context-submit-button" onPress={handleSubmit}>
          Submit
        </Button>
      </FormProvider>
    );
  };
