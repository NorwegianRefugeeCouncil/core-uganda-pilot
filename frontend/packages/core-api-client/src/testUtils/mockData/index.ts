export * from './Form';
export * from './FormField';
export * from './Record';
