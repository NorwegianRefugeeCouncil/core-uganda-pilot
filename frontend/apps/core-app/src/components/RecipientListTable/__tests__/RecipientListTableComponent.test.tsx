import React from 'react';
import { Box } from 'native-base';
import { act, fireEvent, waitFor } from '@testing-library/react-native';

import { render } from '../../../testUtils/render';
import { RecipientListTableComponent } from '../RecipientListTableComponent';

const prepareRowMock = jest.fn();
const clickHandlerMock = jest.fn();
const table = {
  rows: [
    {
      id: 'row1',
      values: {
        recordId: 'recordId',
        col1: 'sdf',
        col2: 'sadf',
      },
      cells: [
        {
          column: 'col1',
          render: jest.fn().mockImplementation((a) => {
            <Box>{a}</Box>;
          }),
          value: 'val1',
          row: 'row1',
          getCellProps: jest.fn(),
        },
        {
          column: 'col2',
          render: jest.fn().mockImplementation((a) => {
            <Box>{a}</Box>;
          }),
          value: 'val2',
          row: 'row1',
          getCellProps: jest.fn(),
        },
      ],
    },
  ],
  columns: [
    {
      id: 'col1',
      Header: 'col1',
      accessor: 'col1',
      hidden: false,
      render: jest.fn().mockImplementation((a) => {
        <Box>{a}</Box>;
      }),
    },
    {
      id: 'col2',
      Header: 'col2',
      accessor: 'col2',
      hidden: true,
      render: jest.fn().mockImplementation((a) => {
        <Box>{a}</Box>;
      }),
    },
  ],
  prepareRow: prepareRowMock,
};

describe('RecipientListTableComponent', () => {
  describe('should match the snapshot', () => {
    it('data', () => {
      const { toJSON } = render(
        <RecipientListTableComponent
          table={table}
          onItemClick={jest.fn()}
          title="title"
          error={null}
          loading={false}
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
    it('error', () => {
      const { toJSON } = render(
        <RecipientListTableComponent
          table={{ rows: [], columns: [], prepareRow: jest.fn() }}
          onItemClick={jest.fn()}
          title="title"
          error="error message"
          loading={false}
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
    it('loading', () => {
      const { toJSON } = render(
        <RecipientListTableComponent
          table={{ rows: [], columns: [], prepareRow: jest.fn() }}
          onItemClick={jest.fn()}
          title="title"
          error={null}
          loading
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should render rows', () => {
    render(
      <RecipientListTableComponent
        table={table}
        onItemClick={clickHandlerMock}
        title="title"
        error={null}
        loading
      />,
    );
    expect(prepareRowMock).toHaveBeenCalledTimes(1);
  });

  it('should call row click handler', () => {
    const { debug, getByTestId } = render(
      <RecipientListTableComponent
        table={table}
        onItemClick={clickHandlerMock}
        title="title"
        error={null}
        loading
      />,
    );
    debug();
    const row = getByTestId('recipient-list-table-row-row1');
    act(() => {
      fireEvent.press(row);
    });
    waitFor(() => expect(clickHandlerMock).toHaveBeenCalledWith('row1'));
  });
});
