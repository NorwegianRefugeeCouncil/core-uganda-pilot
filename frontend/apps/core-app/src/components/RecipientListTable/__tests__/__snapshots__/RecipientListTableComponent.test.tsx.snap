// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipientListTableComponent should match the snapshot data 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "marginBottom": 60,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          dataSet={Object {}}
          level="2"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#000000",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 28,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 34,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          title
        </Text>
        <View
          colorScheme="coolGray"
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#24303E",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 20,
              "marginLeft": 10,
              "marginRight": 10,
              "paddingBottom": 2,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 2,
              "width": 28,
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="5"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fff",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 14,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 17,
                "textAlign": "auto",
                "textDecorationLine": undefined,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      dataSet={Object {}}
      horizontal={true}
      style={Object {}}
    >
      <View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#F6F6F7",
                "borderBottomColor": "#404040",
                "borderBottomWidth": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "justifyContent": "space-between",
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                }
              }
            >
              <div>
                box
              </div>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <View
                _text={
                  Object {
                    "color": "lightText",
                  }
                }
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                colorScheme="secondary"
                dataSet={Object {}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onHoverIn={[Function]}
                onHoverOut={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "borderRadius": 4,
                    "paddingBottom": 4,
                    "paddingLeft": 4,
                    "paddingRight": 4,
                    "paddingTop": 4,
                  }
                }
                testID="recipient-table-sort-button"
              >
                <RNSVGSvgView
                  accessibilityRole="image"
                  align="xMidYMid"
                  bbHeight={20}
                  bbWidth={20}
                  color={4280690214}
                  dataSet={Object {}}
                  fill="currentColor"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  stroke=""
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "color": "#262626",
                        "height": 20,
                        "width": 20,
                      },
                      Object {
                        "flex": 0,
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                  testID="sort-icon-arrowDown"
                  tintColor={4280690214}
                  vbHeight={20}
                  vbWidth={20}
                >
                  <RNSVGGroup
                    fill={
                      Array [
                        2,
                      ]
                    }
                    propList={
                      Array [
                        "fill",
                        "stroke",
                      ]
                    }
                    stroke={null}
                  >
                    <RNSVGGroup>
                      <RNSVGPath
                        d="M14.59 6.29492L10 10.8749L5.41 6.29492L4 7.70492L10 13.7049L16 7.70492L14.59 6.29492Z"
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <RCTScrollView
              contentContainerStyle={
                Array [
                  Object {},
                  Object {
                    "dataSet": Object {},
                  },
                ]
              }
              data={
                Array [
                  Object {
                    "cells": Array [
                      Object {
                        "column": "col1",
                        "getCellProps": [MockFunction],
                        "render": [MockFunction] {
                          "calls": Array [
                            Array [
                              Object {
                                "$$typeof": Symbol(react.memo),
                                "compare": null,
                                "type": Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "render": [Function],
                                },
                              },
                              Object {
                                "children": "val1",
                                "level": "2",
                                "variant": "body",
                              },
                            ],
                          ],
                          "results": Array [
                            Object {
                              "type": "return",
                              "value": <div>
                                box
                              </div>,
                            },
                          ],
                        },
                        "row": "row1",
                        "value": "val1",
                      },
                      Object {
                        "column": "col2",
                        "getCellProps": [MockFunction],
                        "render": [MockFunction] {
                          "calls": Array [
                            Array [
                              Object {
                                "$$typeof": Symbol(react.memo),
                                "compare": null,
                                "type": Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "render": [Function],
                                },
                              },
                              Object {
                                "children": "val2",
                                "level": "2",
                                "variant": "body",
                              },
                            ],
                          ],
                          "results": Array [
                            Object {
                              "type": "return",
                              "value": <div>
                                box
                              </div>,
                            },
                          ],
                        },
                        "row": "row1",
                        "value": "val2",
                      },
                    ],
                    "id": "row1",
                    "values": Object {
                      "col1": "sdf",
                      "col2": "sadf",
                      "recordId": "recordId",
                    },
                  },
                ]
              }
              dataSet={Object {}}
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              style={Object {}}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <View>
                <View
                  onLayout={[Function]}
                  style={null}
                >
                  <View
                    _text={
                      Object {
                        "color": "darkText",
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    dataSet={Object {}}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onHoverIn={[Function]}
                    onHoverOut={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#fff",
                      }
                    }
                    testID="recipient-list-table-row-row1"
                  >
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <View
                        dataSet={Object {}}
                        style={
                          Object {
                            "borderBottomColor": "#C7C7C7",
                            "borderBottomWidth": 1,
                            "flexGrow": 1,
                            "paddingBottom": 8,
                            "paddingLeft": 8,
                            "paddingRight": 8,
                            "paddingTop": 8,
                          }
                        }
                      >
                        <div>
                          box
                        </div>
                      </View>
                      <View
                        dataSet={Object {}}
                        style={Object {}}
                      />
                      <View
                        dataSet={Object {}}
                        style={
                          Object {
                            "borderBottomColor": "#C7C7C7",
                            "borderBottomWidth": 1,
                            "flexGrow": 1,
                            "paddingBottom": 8,
                            "paddingLeft": 8,
                            "paddingRight": 8,
                            "paddingTop": 8,
                          }
                        }
                      >
                        <div>
                          box
                        </div>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RecipientListTableComponent should match the snapshot error 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "marginBottom": 60,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          dataSet={Object {}}
          level="2"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#000000",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 28,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 34,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          title
        </Text>
        <View
          colorScheme="coolGray"
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#24303E",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 20,
              "marginLeft": 10,
              "marginRight": 10,
              "paddingBottom": 2,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 2,
              "width": 28,
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="5"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fff",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 14,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 17,
                "textAlign": "auto",
                "textDecorationLine": undefined,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
    <Text
      dataSet={Object {}}
      level="1"
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#000000",
          "fontFamily": "Roboto_400Regular",
          "fontSize": 18,
          "fontStyle": undefined,
          "fontWeight": undefined,
          "letterSpacing": 0,
          "lineHeight": 26,
          "textAlign": "auto",
          "textDecorationLine": undefined,
        }
      }
    >
      error message
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`RecipientListTableComponent should match the snapshot loading 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "marginBottom": 60,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={Object {}}
    >
      <View
        dataSet={Object {}}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <Text
          dataSet={Object {}}
          level="2"
          style={
            Object {
              "backgroundColor": undefined,
              "color": "#000000",
              "fontFamily": "Roboto_500Medium",
              "fontSize": 28,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 34,
              "textAlign": "auto",
              "textDecorationLine": undefined,
            }
          }
        >
          title
        </Text>
        <View
          colorScheme="coolGray"
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#24303E",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 20,
              "marginLeft": 10,
              "marginRight": 10,
              "paddingBottom": 2,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 2,
              "width": 28,
            }
          }
        >
          <Text
            dataSet={Object {}}
            level="5"
            style={
              Object {
                "backgroundColor": undefined,
                "color": "#fff",
                "fontFamily": "Roboto_500Medium",
                "fontSize": 14,
                "fontStyle": undefined,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 17,
                "textAlign": "auto",
                "textDecorationLine": undefined,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
    </View>
    <View
      dataSet={Object {}}
      fadeDuration={0.1}
      speed={1}
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 40,
          "overflow": "hidden",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#e5e5e5",
            "height": "100%",
            "opacity": 0,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
